The Mac Weekly/utils.swift

```diff
@@ -138,6 +142,81 @@ public class TestableUtils {
         }
         
     }
+    
+    // Verify that a user's email meets our standards (ie is a mac.edu email)
+    static func validateLogin(email: String) -> Bool {
+        
+        let pattern = "^[A-Z0-9._%+-]+@macalester.edu$"
+        // The "^" and "$" characters indicate the start and end of the string, ensuring that the whole string must fit this pattern
+        // ALSO: two slashes are used here instead of one because swift gets weird about regular expressions
+        
+        let doesMatch = email.range(of: pattern, options: [.regularExpression, .caseInsensitive]) != nil
+        
+        return doesMatch
+        
+    }
+    
+    // Log in a user through Firebase
+    static func login(email: String, password: String) {
+        
+        print("About to sign in user")
+        Auth.auth().signIn(withEmail: email, password: password) { (user, error) in
+            
+            if let error = error {
+                print(error)
+            }
+            else {
+                print("User signed in")
+            }
+        }
+    }
+    
+    // Register a new user
+    static func register(email: String, password: String) {
+        
+        if validateLogin(email: email) {
+            
+            // NOTE: this automatically signs in this new user
+            Auth.auth().createUser(withEmail: email, password: password) { (authResult, error) in
+                
+                if let error = error {
+                    
+                    print(error)
+                }
+                else {
+                    // Assuming nothing goes wrong with creating/signing in a new user, send them an email
+                    print("New user created, about to send email verification")
```

Comment:
Need a proper logging system too. Print statements work fine for individual features, but across many features it becomes really unwieldy.